### PR TITLE
Change to display Popup on Active Window page on Windows

### LINKED ISSUE #1616: [BUG] Popup show on unexpected window when app using multiple window
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

There have 2 window in the app, 
When I want to show the popup in the second window, but the popup show on the first window.

### Expected Behavior

There have 2 window in the app, 
When I want to show the popup in the second window, the popup should be shown on the second window.

### Steps To Reproduce

1. open the sln using vs 2022.
2. f5 to run the app
3. click "OpenWin" button on the first window to open the second window.
4. click "OpenPopup" button to show the Popup

### Link to public reproduction project repository

https://github.com/jingliancui/MAUIMultipleWinPopupIssue

### Environment

```markdown
- .NET MAUI CommunityToolkit:8
- OS:Windows 11 19041.0
- .NET MAUI: 8
```

### Anything else?

no

### PR DESCRIPTION
**Pull request by cat0363**
This PR resolves the issue where popups are not displayed on the active window page on Windows.

 ### Description of Change ###

To get the active windows in Windows, look for each window whose IsActivated property is true in Application.Current.Windows.
Call the ShowPopup and ShowPopupAsync methods on the active window page.
Below are the changes.

[src\CommunityToolkit.Maui\PopupService.cs]

    public void ShowPopup<TViewModel>() where TViewModel : INotifyPropertyChanged
    {
        var popup = GetPopup(typeof(TViewModel));

        ValidateBindingContext<TViewModel>(popup, out _);

    #if WINDOWS
        Application.Current!.Windows.First(x => x.IsActivated).Page!.ShowPopup(popup);
    #else
        CurrentPage.ShowPopup(popup);
    #endif
    }

    public void ShowPopup<TViewModel>(TViewModel viewModel) where TViewModel : INotifyPropertyChanged
    {
        ArgumentNullException.ThrowIfNull(viewModel);

        var popup = GetPopup(typeof(TViewModel));

        ValidateBindingContext<TViewModel>(popup, out _);

    #if WINDOWS
        Application.Current!.Windows.First(x => x.IsActivated).Page!.ShowPopup(popup);
    #else
        CurrentPage.ShowPopup(popup);
    #endif
    }

    public void ShowPopup<TViewModel>(Action<TViewModel> onPresenting) where TViewModel : INotifyPropertyChanged
    {
        ArgumentNullException.ThrowIfNull(onPresenting);

        var popup = GetPopup(typeof(TViewModel));

        ValidateBindingContext(popup, out TViewModel viewModel);

        onPresenting.Invoke(viewModel);

    #if WINDOWS
        Application.Current!.Windows.First(x => x.IsActivated).Page!.ShowPopup(popup);
    #else
        CurrentPage.ShowPopup(popup);
    #endif
    }

    public Task<object?> ShowPopupAsync<TViewModel>(CancellationToken token = default) where TViewModel : INotifyPropertyChanged
    {
        var popup = GetPopup(typeof(TViewModel));

        ValidateBindingContext<TViewModel>(popup, out _);

    #if WINDOWS
        return Application.Current!.Windows.First(x => x.IsActivated).Page!.ShowPopupAsync(popup, token);
    #else
        return CurrentPage.ShowPopupAsync(popup, token);
    #endif
    }

    public Task<object?> ShowPopupAsync<TViewModel>(TViewModel viewModel, CancellationToken token = default) where TViewModel : INotifyPropertyChanged
    {
        ArgumentNullException.ThrowIfNull(viewModel);

        var popup = GetPopup(typeof(TViewModel));

        ValidateBindingContext<TViewModel>(popup, out _);

    #if WINDOWS
        return Application.Current!.Windows.First(x => x.IsActivated).Page!.ShowPopupAsync(popup, token);
    #else
        return CurrentPage.ShowPopupAsync(popup, token);
    #endif
    }

 

 ### Linked Issues ###
 

 - Fixes #1616

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

 
 
Below are the verification results.

https://github.com/CommunityToolkit/Maui/assets/125236133/ccaadede-f21a-4351-b6d2-a3c53a1339ee

You can see that the Popup is displayed in the active window.

### CODE REVIEW

**Comment by cat0363** [comment]
In cases where a popup is displayed by setting a timer, it will not be displayed intentionally. . .
In such cases, the active window may change from the Popup's caller until the Popup is displayed.
If you need to consider such cases, please discard this PR.

**Comment by cat0363** [comment]
It would be best if there was a way to determine which page is calling the Popup. . .

**Comment by cat0363** [comment]
Perhaps to solve this problem we need some variation of Popup's ShowPopup, ShowPopupAsync methods that pass an instance of the page we want to display.
If you use the current page, the problem I previously described will occur.

File: src/CommunityToolkit.Maui/PopupService.cs [hunk 1]
```diff
@@ -52,7 +52,18 @@ public PopupService()
 
 		ValidateBindingContext<TViewModel>(popup, out _);
 
+#if WINDOWS
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/PopupService.cs:55]: can you extract it to the single method and call only once? also could you please remove all null-forgiving operators. they are forbidden in this repo

**Comment by bijington** [comment]
I am just wondering if we could opt for registering the `PopupService` as scoped given .NET MAUI scopes per window, then we could somehow obtain the active window at the time of resolution and rely on that through the life of the `PopupService` instance. What do people think?

**Review by brminnick — APPROVED** [review]

**Pull request merged**